### PR TITLE
packaging: add type hints and drop support for file parameter in bytes type

### PR DESCRIPTION
**Pull request by bdrung**
Add type hints to most methods of `PackageInfo` and its subclasses.

The method `_search_contents` is only called by `get_file_package` and `get_file_package` requires the `file` parameter to be of type `str`. Otherwise the `uninstalled=False` case would fail.

So drop supporting `bytes` in `_search_contents`.

### CODE REVIEW

**Review by schopin-pro — COMMENTED** [review]
I have some questions and suggestions.

File: apport/packaging.py [hunk 1]
```diff
@@ -317,7 +319,7 @@ def get_uninstalled_package(self):
                 continue
             except ValueError:
                 return package
-        return None
+        raise ValueError
```
[the remark this answers is not in the document]
Comment (bdrung) [reply apport/packaging.py:329]: This code path is currently not hit. Then the caller does not have to check the `None` case.

File: apport/packaging_impl/apt_dpkg.py [hunk 2]
```diff
@@ -1106,10 +1106,10 @@ def install_packages(
                     # but as policy states it is invalid to use that in
                     # Replaces/Depends, we can safely choose the first value
                     # here.
-                    conflict = conflict[0]
-                    if apt_cache.is_virtual_package(conflict[0]):
+                    conflict0 = conflict[0]
```
[the remark this answers is not in the document]
Comment (bdrung) [reply apport/packaging_impl/apt_dpkg.py:1115]: Changed. I carved that code out into the separate function `_remove_conflicting_packages` and unpacked `conflict[0]`. That approach is more readable.

File: apport/packaging.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (schopin-pro) [reply apport/packaging.py:329]: Please document that reason in the commit log.

[the remark this answers is not in the document]
Comment (bdrung) [reply apport/packaging.py:329]: Done.

**Pull request merged**